### ISSUE TITLE
OCIOLut plugin not working in latest Windows binary

### ISSUE THREAD

**Issue by diaasami** [tuttleofx/TuttleOFX#418]
Using the latest binary from http://www.tuttleofx.org/download on Windows, OCIOLut plugin doesn't work, it used to work with the version before that but with the latest it doesn't.

I tried using the free Adobe AfterEffects plugin based on OCIO and it works, so it seems something with the latest build.

Any idea why that might be ?

**Comment by diaasami** [comment]
I have just tried sam to do the same thing and it seems to work, which is surprising.

**Comment by cchampet** [comment]
Hi,
So we are looking for something different between Tuttle and Vegas hosts, about openFX plugins.
Do you have an error message from Vegas?

**Comment by diaasami** [comment]
Yes it seems so, I've just tried the TuttleBlur plugin and it's the same.

There are no obvious error messages, I went and checked Vegas logs but there are no errors, the logs from both tuttleofx and vegas look OK, I noticed that in the preview thumbnail in Vegas, which shows an image split in half, with the left part untouched and the right part with the plugin applied, the right half shows up as a solid mustard color.

Tell me if you have any ideas, I'll probably try debugging this later today.

**Comment by diaasami** [comment]
Turns out that Vegas passes the input to the plugin with UByte bit depth, although the plugin clearly mentions that only float is supported during initialization.

Is this common ? do you have other OFX hosts misbehaving like this ?

**Comment by cchampet** [comment]
This is a strange behavior, and I am not aware of other OFX hosts which behave like this.
Like you said, the OCIOLut plugin declares that [only float is supported](https://github.com/tuttleofx/TuttleOFX/blob/develop/plugins/image/process/color/OCIO/src/OCIOLut/OCIOLutPluginFactory.cpp#L36).
Did you try to contact Sony Vegas guys?
In your first message, you told us that the previous version of the plugin worked. Was it with the same version of Vegas?

**Comment by diaasami** [comment]
I dug in the history and I found that there was an [old version of OCIO](https://github.com/tuttleofx/TuttleOFX/commit/d6c3e7ef10329ae6813ba937759f227f822990e3) that supported all bitdepths and that's the one that worked I believe.

**Comment by cchampet** [comment]
Ok I don't know the reason of this changes.
@fabiencastan ?

**Comment by diaasami** [comment]
Let me add more information:
1. You can see how it looks, compared to other plugins, ![here](https://cloud.githubusercontent.com/assets/4257328/7812568/bce3e888-03b3-11e5-8dd0-068c1b37222f.png), the exact same happens with tuttleBlur.
2. I have just tried another [non-tuttle OFX plugin](http://asawicki.info/news_1579_writing_ofx_video_processing_plugin.html) that supports only 8bit depth and it works in Vegas.
3. I tried a few other compositors:
    1. DaVinci Resolve 12: Doesn't work.
    2. Natron v1.2.1: Works.

**Comment by archetyped** [comment]
Hi, I would love to get TuttleOFX plugins working in Sony Vegas Pro as well.  Unfortunately, I'm not even seeing them in Vegas' Video FX tab when I add them to Vegas' OFX plugins directory.

Any instructions for using TuttleOFX plugins in an OFX-compatible host such as Sony Vegas Pro would be greatly appreciated.

Thanks.

**Comment by cchampet** [comment]
Hello,
What is your platform (Mac OS, Win...) ?
Do you have some log from Sony Vegas Pro ?

The first step to use any OpenFX plugin in any OpenFX host, is to set an environment variable to tell where are the plugins. The variable is __OFX_PLUGIN_PATH__. After that, each host has its specificities and its ways to manage OpenFX plugins. 

But you are right we will describe a bit more how to use TuttleOFX plugins into each famous host.

**Comment by archetyped** [comment]
Hi, apologies for not clarifying, Vegas Pro is Windows only.

**Comment by archetyped** [comment]
Hi, this may have some information regarding Vegas-specific OFX compatibility: https://github.com/devernay/openfx-vegas

**Comment by diaasami** [comment]
Based on my investigation and work, which I can't share unfortunately, there are two problems with Tuttle working in Vegas(please take this with a grain of salt as I am new to OpenFX):
# Vegas not providing multithreading support functions, e.g. mutexCreate always returns an error, which causes Tuttle to crash in initialization, I fixed this by replacing calls to mutexCreate with the new C++11 mutex class.
# Vegas is probably violating OpenFX standard here, it passes image formats that are not supported by the plugin regardless of the that plugin declares during the describe phase, so in my case, because the default for projects in Vegas is 8-bit, it passed 8-bit data to the plugin although the plugin declares that it only supports 32-bit data, this can be fixed by changing the project bitdepth to 32-bit.

After doing these or changes. I was able to get Tuttle plugins working in Vegas.

**Comment by cchampet** [comment]
Thank for the great job you did @diaasami !
We will do the corresponding updates to use Tuttle plugins on Vegas. But until this work be done, could you share with us your build of Tuttle which works on Vegas?

**Comment by diaasami** [comment]
release 0.8

**Comment by archetyped** [comment]
I've downloaded [release 0.8](https://github.com/tuttleofx/TuttleOFX/releases/tag/v0.8) and can confirm that the OFX plugins are indeed visible in Sony Vegas.

However, while you can add TuttleOFX's plugins to a video clip, none of the plugins have any effect on the video's image.

**Comment by archetyped** [comment]
Quick Update: Just came across [Vision Color's OFX LUT Plugin](http://www.vision-color.com/lutplugin/), which is based on TuttleOFX's OCIOLut plugin.  Perhaps this might provide some insight on what needs to be done for the official TuttleOFX plugins to work in Vegas.

![vision-color_ofx-lut-plugin_about](https://cloud.githubusercontent.com/assets/3237917/9421084/002c59a6-4809-11e5-8977-bbb3c7d3391c.png)
